### PR TITLE
Create todo app

### PR DESCRIPTION
**Pull request by ghulamhafeez**
## Todo 
- Create todo list app with functional component

### CODE REVIEW

File: src/views/Todo-Components/TodoForm.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import TodoList from "./TodoList";
+const TodoForm = () => {
+  const [todo, setTodo] = useState([]);
+  const [todoo, setTodoo] = useState();
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: change variable name

File: src/views/Todo-Components/TodoForm.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import TodoList from "./TodoList";
+const TodoForm = () => {
+  const [todo, setTodo] = useState([]);
+  const [todoo, setTodoo] = useState();
+  const [showButton, setShowButton] = useState(false);
+  const [index, setIndex] = useState();
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: no need of index

File: src/views/Todo-Components/TodoForm.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import TodoList from "./TodoList";
+const TodoForm = () => {
+  const [todo, setTodo] = useState([]);
+  const [todoo, setTodoo] = useState();
+  const [showButton, setShowButton] = useState(false);
+  const [index, setIndex] = useState();
+
+  console.log("e", todoo);
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: please remove the console

File: src/views/Todo-Components/TodoForm.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const TodoForm = () => {
+  const [todo, setTodo] = useState([]);
+  const [todoo, setTodoo] = useState();
+  const [showButton, setShowButton] = useState(false);
+  const [index, setIndex] = useState();
+
+  console.log("e", todoo);
+  const handleAdd = () => {
+    let myData = {
+      name: todoo,
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: add a property of id

File: src/views/Todo-Components/TodoForm.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    };
+    console.log("e");
+    const newTodo = todo.slice();
+    console.log("newTodo", newTodo);
+    newTodo.push(myData);
+    setTodo(newTodo);
+    handleNull();
+  };
+
+  const handleDelete = (index) => {
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: change the variable name handleDelete to isDeleted

File: src/views/Todo-Components/TodoForm.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    const newTodo = todo.slice();
+    newTodo.splice(index, 1);
+    setTodo(newTodo);
+  };
+
+  const handleEdit = (data, index) => {
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: change the variable name

File: src/views/Todo-Components/TodoForm.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    setIndex(index);
+    setTodoo(data.name);
+    setShowButton(true);
+  };
+
+  const handleUpdate = () => {
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoForm.js]: please change the variable name

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import TodoForm from "../src/views/Todo-Components/TodoForm";
+const App = () => {
+  return (
+    <div>
+      <TodoForm />
```
Comment (chusama-dev) [inline src/App.js]: make a component Todos

File: src/views/Todo-Components/TodoList.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+const TodoList = ({ todo, onEdit, onDelete }) => {
+  return (
+    <div>
+      <table className="table">
+        <tr className="th">
+          <th>Names</th>
+          <th>Buttons</th>
+        </tr>
+        <tr className="th">
+          <ul>
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoList.js:12]: remove ul

File: src/views/Todo-Components/TodoList.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+            {todo.map((data, i) => {
+              return (
+                <li key={data.id}>
```
Comment (chusama-dev) [inline src/views/Todo-Components/TodoList.js:15]: use tr not li

File: src/views/Todo-Components/Todos.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import TodoList from "./TodoList";
+import TodoForm from "./TodoForm";
+const Todos = () => {
+  const [todoList, setTodoList] = useState([]);
```
Comment (chusama-dev) [inline src/views/Todo-Components/Todos.js]: chnge the name of todoList , Todos

File: src/views/Todo-Components/Todos.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+  const onEdit = (data, i) => {
+    setTodo(data.name);
+    setIndex(i);
+    setIsEditMode(true);
+  };
+
+  const onUpdate = () => {
+    const newTodo = todoList.slice();
+    let data = {
```
Comment (chusama-dev) [inline src/views/Todo-Components/Todos.js]: Use findIndex

**Comment by Monfernape** [comment]
Good review @Hawakeyes

**Pull request merged**